altair/api.py

```diff
@@ -409,6 +409,70 @@ def to_dict(self):
         D['config'] = self.vlconfig.to_dict()
         return D
 
+    def select_x(self):
+        """
+        Helper function that does a best effort of selecting an automatic x axis.
+
+        Raises AssertionError if it cannot find a x axis to choose.
+        """
+        assert self.data is not None
+        assert len(self.data) >= 1
+
+        d = self._classify_data_by_type()
+
+        # Choose the first column found on the following order: T, O, N, Q
+        chosen_x = None
+        for typ in ['Q', 'N', 'O', 'T']:
+            if len(d[typ]) >= 1:
+                chosen_x = d[typ][0]
+
+        if chosen_x is None:
+            raise AssertionError('Could not select X.')
```

Comment:
Aside from the code level details that Jake has brought up. I am not sure
that it makes sense to try and autoselect X and Y. In complex, multi-column
datasets, it is hard to imagine that will produce anything useful.

On Wed, Oct 21, 2015 at 7:06 PM, Jake Vanderplas <notifications@github.com>
wrote:

> In altair/api.py
> <https://github.com/ellisonbg/altair/pull/78#discussion_r42694662>:
>
> > +
> > +        Raises AssertionError if it cannot find a x axis to choose.
> > +        """
> > +        assert self.data is not None
> > +        assert len(self.data) >= 1
> > +
> > +        d = self._classify_data_by_type()
> > +
> > +        # Choose the first column found on the following order: T, O, N, Q
> > +        chosen_x = None
> > +        for typ in ['Q', 'N', 'O', 'T']:
> > +            if len(d[typ]) >= 1:
> > +                chosen_x = d[typ][0]
> > +
> > +        if chosen_x is None:
> > +            raise AssertionError('Could not select X.')
>
> Or perhaps:
>
> from itertools import chaintry:
>     chosen_x = next(chain(*(d[typ] for typ in ['Q', 'N', 'O', 'T'])))except StopIteration:
>     raise ValueError('Could not select X.')
>
> but maybe now I'm just getting a little over-zealous...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ellisonbg/altair/pull/78/files#r42694662>.
>



-- 
Brian E. Granger
Associate Professor of Physics and Data Science
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com